ui-v2/app/components/templated-anchor.js

```diff
@@ -0,0 +1,68 @@
+import Component from '@ember/component';
+import { get, set, computed } from '@ember/object';
+
+const createWeak = function(wm = new WeakMap()) {
+  return {
+    get: function(ref, prop) {
+      let map = wm.get(ref);
+      if (map) {
+        return map[prop];
+      }
+    },
+    set: function(ref, prop, value) {
+      let map = wm.get(ref);
+      if (typeof map === 'undefined') {
+        map = {};
+        wm.set(ref, map);
+      }
+      map[prop] = value;
+      return map[prop];
+    },
+  };
+};
+const weak = createWeak();
+
+const templateRe = /{{([A-Za-z.]*)}}/g;
```

Comment:
Do you need to escape `{` and `}` in JS regexp? I guess not if this works just wondering if it's one of those things where not escaping seems to work but accidentally matches other stuff like how `foo.com` matches `fooZcom`? Other examples I've seen of handlebars syntax regexp in JS do escape the opening ones at least?

Reply:
Not that I'm aware of, it's potentially one of those 'optional' escapes that eslint warns you that it's not needed, will take a look though. First glance its looks like `\{` and `{` both match the same thing.

Reply:
Ok so, if I've understood correctly, generally you do not need to escape `{`s unless it's part of a repetition operator (like `{1,3}`) - but you can if you want. I think I'll put some comments above this at least though. I usually do that with RegExp to make them super clear.

Actually, the more I look at this the more I wonder, I should probably switch the `*` to a `?`. Plus, do you know whether we are allowing people to use meta data for interpolation, I was trying to keep the matches pretty restrictive just for our API keys but maybe I should add `0-9` and `_-` in there?

> Key can contain only ASCII chars and no special characters (A-Z a-z 0-9 _ and -)

What do you think?

Reply:
Restrictive is great... see reply in main conversation.